### ISSUE TITLE
是否使用了无标签数据的gt？

### ISSUE THREAD

**Issue by SII-Whaleice** [azhuantou/HSSDA#16]
您好，感谢您的工作，在阅读您的代码后我有一个问题:

我发现在pvrcnn_ssl.py中，您在filter_middle_pseudo_label函数中有两处使用了gt_boxes。
`# iou_flag to filter out pseudo-labels that overlap with gt
            if len(train_box) != 0:
                iou = iou3d_nms_utils.boxes_bev_iou_cpu(train_box, test_box)
                iou_value = np.sum(iou, axis=0)

                iou_flag = iou_value == 0
            else:
                iou_flag = np.array([True] * len(test_box))`
以及
`ori_gt_box = batch_dict['gt_boxes']
        new_gt_box_list = []
        for bs_idx in range(batch_size):
            cur_gt_box = ori_gt_box[bs_idx]
            cur_gt_box = cur_gt_box[torch.sum(cur_gt_box, 1) != 0]
            if label_mask[bs_idx]:   # labeled scene do not mine pseudo-label
                new_gt_box_list.append(cur_gt_box)
                continue

            if len(cur_gt_box) > 0:
                cur_gt_box = [bb.unsqueeze(0) for bb in cur_gt_box if bb[0] != 0]
                cur_gt_box = torch.cat(cur_gt_box) if len(cur_gt_box) > 0 else torch.zeros((0, 8)).to(ori_gt_box.device)
                cur_gt_box = cur_gt_box if len(cur_gt_box.shape) > 1 else cur_gt_box.unsqueeze(0)
                cur_pseudo_box = select_mid_box[bs_idx]
                new_gt_box = torch.cat((cur_gt_box, cur_pseudo_box), dim=0)
                new_gt_box_list.append(new_gt_box)`
想询问您是出于什么原因这么做，据我所知在半监督设置下不能使用无标签数据的label，期待您的回复，谢谢您！

**Comment by azhuantou** [comment]
您好，感谢您的关注！

按照我们在README中提供的数据处理流程，您所说的“gt_boxes”并不包含任何无标签数据的gt信息，也不涉及到使用无标签数据的label，谢谢！

**Comment by dof-pikes** [comment]
我跟这位同行有同样的疑惑，所以这行代码中的cur_gt_box 包含的是什么信息呢？https://github.com/azhuantou/HSSDA/blob/main/pcdet/models/detectors/pv_rcnn_ssl.py#L507
您好像并没有在这里区分该样本是有标签还是无标签数据，换句话说，该样本是无标签数据会cat cur_gt_box 和 cur_pseudo_box。
根据我的理解，您在https://github.com/azhuantou/HSSDA/blob/main/pcdet/models/detectors/pv_rcnn_ssl.py#L407 这一段代码里是计算了预测的box和GT之间的IoU，然后用这个IoU_flag筛选除GT_box之外的预测结果（KITTI数据集标签是不完整的），也就是cur_pseudo_box。如果我的理解有误，请您指出~感谢回复！

**Comment by dof-pikes** [comment]
懂了，cur gt 是前一个”interval“ 产生的”hard“伪标签。

另外，为啥要mask Ped，直接屏蔽Ped这一类，不是变成了单阈值吗？这样Ped要么当作"gt_box"，要么被移除。感谢您的回复！

**Comment by SII-Whaleice** [comment]
谢谢，我明白了！
